### PR TITLE
[BEAM-1842] Add CreateViewWithViewFn PTransformMatcher

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---   
This matcher is to stop matching the composite View.asFoo PTransforms,
as a PipelineRunner should support materialization of arbitrary
PTransforms created with the CreatePCollectionView transform.

### CODE REVIEW

**Comment by tgroh** [comment]
retest this please

**Comment by tgroh** [comment]
R: @jkff OR @kennknowles 

CC: @tweise @aviemzur @aljoscha 

This minorly modifies the composite graph structure for all runners that used to override `View.asFoo` - now the override is `CreatePCollectionView.of(view)` such that `view.getViewFn()` returns a `FooViewFn`

**Comment by tgroh** [comment]
run apex validatesrunner

**Comment by tgroh** [comment]
run dataflow validatesrunner

**Comment by tgroh** [comment]
run spark validatesrunner

**Comment by tgroh** [comment]
run flink validatesrunner

**Comment by kennknowles** [comment]
The approach seems right. Moving in this direction will allow views to be supported in the new way. LGTM if the tests pass.

**Comment by tgroh** [comment]
run spark validatesrunner

**Comment by tgroh** [comment]
run apex validatesrunner

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformMatchers.java [hunk 1]
```diff
@@ -168,6 +171,21 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
     };
   }
 
+  public static PTransformMatcher createViewWithViewFn(final Class<? extends ViewFn> viewFnType) {
```
Comment (aviemzur) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformMatchers.java]: The PR is already merged, but could be valuable to add a javadoc for this method.

**Pull request closed without merging**